src/openeo_gfmap/fetching/commons.py

```diff
@@ -76,19 +79,53 @@ def filter_condition(band_name, _):
     )
 
 
+def _load_collection_hybrid(
+    connection: openeo.Connection,
+    is_stac: bool,
+    collection_id_or_url: str,
+    bands: list,
+    spatial_extent: Union[Dict[str, float], Parameter, None] = None,
+    temporal_extent: Union[Sequence[InputDate], Parameter, str, None] = None,
+    properties: Optional[dict] = None,
+):
+    """Wrapper around the load_collection, or load_stac method of the openeo connection."""
+    if not is_stac:
+        return connection.load_collection(
+            collection_id=collection_id_or_url,
+            spatial_extent=spatial_extent,
+            temporal_extent=temporal_extent,
+            bands=bands,
+            properties=properties,
+        )
+    cube = connection.load_stac(
+        url=collection_id_or_url,
+        spatial_extent=spatial_extent,
+        temporal_extent=temporal_extent,
+        bands=bands,
+        properties=properties,
+    )
+    print(bands)
+    cube = cube.rename_labels(dimension="bands", target=bands)
+    return cube
+
+
 def _load_collection(
```

Comment:
As this function is getting longer (which is why I get the creation of _load_collection_hybrid) we might consider splitting of the pre and post processing of the different fetchtypes. That way it might be easier to find the differences both as a user and future developer. 
Not sure if this is in the scope of this PR however, we could create an issue for this.